File: frontend/src/app/components/countries/country.ts
import {Region, RegionInterface} from "../regions/region";


export interface CountryInterface {
  countryID: string;
  countryName: string;
  regionID: RegionInterface;
}

export class Country {
  constructor(fields: Partial<CountryInterface>) {
    this.countryID = fields.countryID;
    this.countryName = fields.countryName;
    this.regionID = new Region(fields.regionID);
  }
  countryID: string;
  countryName: string;
  regionID: Region;

}

File: src/main/java/pl/weronika/kurczyna/model/entity/Country.java
package pl.weronika.kurczyna.model.entity;

import org.springframework.lang.Nullable;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.ManyToOne;

@Entity(name = "countries")
public class Country {
    @Id
    @Column(unique = true)
    private String countryID;
    @Nullable
    private String countryName;
    @Nullable
    @ManyToOne
    private Region regionID;


    public Country() {
    }


    @Nullable
    public String getCountryName() {
        return countryName;
    }

    public void setCountryName(@Nullable String countryName) {
        this.countryName = countryName;
    }

    @Nullable
    public Region getRegionID() {
        return regionID;
    }

    public void setRegionID(@Nullable Region regionID) {
        this.regionID = regionID;
    }
}
